### ISSUE TITLE
WAN DHCP Interfaces dont get ARP resolution of auto-detect WAN-gateway

### ISSUE THREAD

**Issue by mitnicki** [opnsense/core#6846]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [ ] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [ ] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

When we get a dhcp address on a WAN Interface we always get 1.2.3.4/32 masks by DHCP Server.
When that happens we are able to configure the option "Far Gateway" for the Gateway. That should be enough for the interface to come up. But its not working. Far Gateway Option only works when we set interface IP to static IPV4 and configure it the same 1.2.3.4/32.

What we can see is that the opnsense doesnt get the arp resolution for the DHCP Gateway when having a Mask of /32 via DHCP.

in the logs we can see:
arpresolve: can't allocate llinfo for <network ip> on OPT1

furthermore we can see that there is no arp entry for the auto-detected gateway resulting in no connectivity when that interface is configured as DHCP Client.

Could you please change the behavior ? The DHCP Interface with mask /32 must be able to do a arp request for the Gateway which is not in the /32 Network of dhcp client address.. .

Thank you!

Software version used and hardware type if relevant, e.g.:

OPNsense latest version as of writing
Hosted on KVM
Virtual Appliance

**Comment by fichtner** [comment]
I don’t see any logs but assuming you have multi-WAN (multiple default gateways) it may already be fixed in the development version.

**Comment by mitnicki** [comment]
yes thats exactly what we are doing. sorry i forgot to mention that its a multi wan setup.

**Comment by fichtner** [comment]
You could try this patch: https://github.com/opnsense/core/commit/c8a5d32760

# opnsense-patch c8a5d32760


Cheers,
Franco

**Comment by mitnicki** [comment]
i will give it a try immediately and report back. thank you so far!

Gesendet von Outlook für iOS<https://aka.ms/o0ukef>
________________________________
Von: Franco Fichtner ***@***.***>
Gesendet: Thursday, September 14, 2023 8:09:40 AM
An: opnsense/core ***@***.***>
Cc: mitnicki ***@***.***>; Author ***@***.***>
Betreff: Re: [opnsense/core] WAN DHCP Interfaces dont get ARP resolution of auto-detect WAN-gateway (Issue #6846)


You could try this patch: c8a5d32760<https://github.com/opnsense/core/commit/c8a5d32760>

# opnsense-patch c8a5d32760


Cheers,
Franco

**Comment by mitnicki** [comment]
Thats awesome. it just works right away now. now im also able to use the gateway monitoring for the additional wan interfaces. Furthermore im able to use your patch on version 23.7 and also on version 24.1 (developer).

**Comment by fichtner** [comment]
Careful with the patch on the development version. You may be removing it because it's already included. ;)


Cheers,
Franco

**Comment by fichtner** [comment]
Note to self: since there wasn't a ticket before use this one. It's marked bug but the scope is quite small. This needs a bit more time before being merged to 23.7.x.

**Comment by fichtner** [comment]
Yeah, the release state is enforced scrubbing patches applied that are not part of the release. E.g. 23.7.4 today will remove the patch again so you need to reapply. I'm trying to get the patch into 23.7.5 but I'm not sure yet.


Cheers,
Franco

**Comment by gionag** [comment]
> You could try this patch: [c8a5d32760](https://github.com/opnsense/core/commit/c8a5d32760)
> 
> ```
> # opnsense-patch c8a5d32760
> ```
> 
> Cheers, Franco

for what is worth, similar problem of the OP, patch solved the problem also for me.
i suggest to promote it

**Comment by fichtner** [comment]
Yeah, as I said it’s on the table for inclusion in 23.7.5.

**Comment by gionag** [comment]
thanks for that,
i was just highlighting the effectiveness of the patch.
